route-rs-runtime/src/link/blackhole_link.rs

```diff
@@ -18,15 +12,29 @@ impl<Packet: Sized> Future for BlackHoleLink<Packet> {
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
         loop {
-            let input_packet_option: Option<Packet> = try_ready!(self.input_stream.poll());
-            match input_packet_option {
-                None => return Ok(Async::Ready(())),
-                Some(_) => {}
-            };
+            for ingress_stream in self.ingress_streams.iter_mut() {
+                if try_ready!(ingress_stream.poll()).is_none() {
+                    return Ok(Async::Ready(()));
+                }
+            }
         }
     }
 }
 
+impl<Packet: Sized + 'static> Link<Packet, ()> for BlackHoleLink<Packet> {
+    fn new(ingress_streams: Vec<PacketStream<Packet>>) -> Self {
+        BlackHoleLink { ingress_streams }
+    }
+
+    fn runnables(self) -> Vec<TokioRunnable> {
+        vec![Box::new(self)]
```

Comment:
I'd just like to pass out `self`, but running into the `doesn't have a size known at compile-time` stuff.

Reply:
Makes sense size is not known cause what we are in this case is a Vec<PacketStream<Packet>>.  Boxing it up is fine. I mean, kind of makes sense all our static lifetimed shit lives on the heap anyways.